sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -877,6 +878,69 @@ case class GpuFromUnixTime(
   override lazy val resolved: Boolean = childrenResolved && checkInputDataTypes().isSuccess
 }
 
+
+class FromUTCTimestampExprMeta(expr: FromUTCTimestamp,
+                               override val conf: RapidsConf,
+                               override val parent: Option[RapidsMeta[_, _, _]],
+                               rule: DataFromReplacementRule)
+  extends BinaryExprMeta[FromUTCTimestamp](expr, conf, parent, rule) {
+
+  override def tagExprForGpu(): Unit = {
+    val timezoneShortID = GpuOverrides.extractStringLit(expr.right).getOrElse("")
+
+    if (timezoneShortID == null) {
+      willNotWorkOnGpu("input time zone is null")
+    }
+
+    val utc = ZoneId.of("UTC").normalized
+    try {
+      // This is copied from Spark, to convert `(+|-)h:mm` into `(+|-)0h:mm`.
+      val timezone = ZoneId.of(timezoneShortID.replaceFirst("(\\+|\\-)(\\d):", "$10$2:"),
+        ZoneId.SHORT_IDS).normalized
+
+      if(!timezone.equals(utc)) {
+        willNotWorkOnGpu("only timezones equivalent to UTC are supported")
+      }
+    } catch {
+      case _: Throwable => willNotWorkOnGpu("invalid input time zone")
```

Comment:
I'm not sure if we need to re-throw the exception here?

Reply:
I don't think we want to catch an exception here.  If the timezone is invalid we should just let it throw, as that's what Spark CPU would do, correct?

Reply:
Yes, Spark CPU doesn't catch anything. So I'm going to remove this catch too.